helix-core/src/syntax.rs

```diff
@@ -294,8 +304,20 @@ impl Loader {
             });
 
         configuration_id.and_then(|&id| self.language_configs.get(id).cloned())
+    }
 
-        // TODO: content_regex handling conflict resolution
+    pub fn language_config_for_shebang(&self, path: &Path) -> Option<Arc<LanguageConfiguration>> {
+        // Read the first 128 bytes of the file. If its a shebang line, try to find the language
+        let file = File::open(path).ok()?;
+        let mut buf = String::with_capacity(128);
+        Read::read_to_string(&mut Read::take(file, 128), &mut buf).ok()?;
```

Comment:
why not just `file.take(128).read_to_string(&mut buf)`? It's a trait

Reply:
I was trying to avoid unwraps that can cause panic, thats why i used if let on those places.
Is it ok to have those here?

Reply:
Yeah no unwraps, just `Read::read(buf).ok()` is equivalent to `buf.read().ok` -- Read is a trait